docs/root/start/sandboxes/grpc-s2s.rst

```diff
@@ -0,0 +1,229 @@
+.. _install_sandboxes_grpc_s2s:
+
+gRPC Service to Service
+=======================
+
+.. sidebar:: Requirements
+
+   .. include:: _include/docker-env-setup-link.rst
+
+This example demonstrates Envoy's support for routing
+`gRPC traffic <https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#config-route-v3-routematch>`__
+and integrating `gRPC health checks <https://github.com/grpc/grpc/blob/master/doc/health-checking.md>`__
+with Envoy's active health checking for upstream clusters using
+`GrpcHealthCheck <https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/health_check.proto#envoy-v3-api-msg-config-core-v3-healthcheck-grpchealthcheck>`__.
+
+The sandbox consists of two gRPC services, ``Hello`` and ``World``. Each service implements a Unary-Unary
+gRPC method, ``Greet()`` as described in :download:`hello.proto <_include/grpc-s2s/protos/hello.proto>`
+and :download:`world.proto <_include/grpc-s2s/protos/world.proto>`.
+
+We run 2 instances of each service using docker-compose's support for running a service
+in `replicated mode <https://docs.docker.com/compose/compose-file/deploy/#replicas>`__.
+
+Two separate Envoy instances, `envoy-hello` and `envoy-world` are configured as a proxy for both services.
+
+The request flow looks as follows for this sandbox:
+
+``[client](gRPC) -> [envoy-hello] -> [hello](2 instances) -> [envoy-world] -> [world](2 instances)``
```

Comment:
Could you help explain why we need to use this model to present the behavior of [gRPC services](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/other_protocols/grpc#grpc-services) instead of using the simple model like 
```
[downstream](gRPC) -> envoy -> [upstream][gRPC]
```
and we can also use 2 instances to present the gRPC healthcheck